crypto/key_derivation.go

```diff
@@ -0,0 +1,49 @@
+package crypto
+
+import (
+	"github.com/bifurcation/mint"
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+)
+
+const (
+	clientExporterLabel = "EXPORTER-QUIC client 1-RTT Secret"
+	serverExporterLabel = "EXPORTER-QUIC server 1-RTT Secret"
+)
+
+// MintState contains
+type MintState interface {
+	GetCipherSuite() mint.CipherSuiteParams
+	ComputeExporter(label string, context []byte, keyLength int) ([]byte, error)
+}
+
+// DeriveAESKeys derives the AES keys and creates a matching AES-GCM AEAD instance
+func DeriveAESKeys(ms MintState, pers protocol.Perspective) (AEAD, error) {
+	var myLabel, otherLabel string
+	if pers == protocol.PerspectiveClient {
+		myLabel = clientExporterLabel
+		otherLabel = serverExporterLabel
+	} else {
+		myLabel = serverExporterLabel
+		otherLabel = clientExporterLabel
+	}
+	myKey, myIV, err := computeKeyAndIV(ms, myLabel)
+	if err != nil {
+		return nil, err
+	}
+	otherKey, otherIV, err := computeKeyAndIV(ms, otherLabel)
+	if err != nil {
+		return nil, err
+	}
+	return NewAEADAESGCM(otherKey, myKey, otherIV, myIV)
+}
+
+func computeKeyAndIV(ms MintState, label string) (key, iv []byte, err error) {
+	cs := ms.GetCipherSuite()
```

Comment:
Should we check here that this is actually AES-GCM?

Reply:
We probably should do that. That will mean checking the cipher suite against the values defined in https://github.com/bifurcation/mint/blob/master/common.go#L43-L52 though, I don't see a general way to check for AES.

Reply:
Actually, what we should do is set a list of supported ciphers in the `mint.Config`, and only use AES ciphers there. We can then trust mint to negotiate one of them.